upgrader.go

```diff
@@ -239,14 +239,17 @@ func (u *Upgrader) UpgradeCluster() error {
 		return err
 	}
 
-	if !isNewer && !u.forceReplacement {
-		u.logger.Println("Upgrade not needed, cluster is already running latest AMI")
+	oldImageFound, err := u.checkRunningInstances(*latestImage.ImageId)
+	if err != nil {
+		return err
+	}
+
+	if !oldImageFound && !isNewer && !u.forceReplacement {
+		u.logger.Println("Upgrade not needed, cluster is already running the latest AMI")
 		return u.terminateOrphanedInstances(asgName)
 	}
 
-	if u.forceReplacement {
-		u.logger.Println("Cluster already running latest AMI, but replacing instances anyway")
-	} else {
```

Comment:
Because of the weird relationship between isNewer and isOldImageFound, perhaps else is fine?

Reply:
Can't have an `else` without an `if`